app/processing/calculation/get-calculation.js

```diff
@@ -1,20 +1,27 @@
 const schema = require('./calculation-schema')
 
 const getCalculationByPaymentRequestId = require('./get-calculation-by-payment-request-id')
+const updateCalculationPaymentRequestId = require('./update-calculation-payment-request-id')
 
 const getCalculation = async (paymentRequestId) => {
-  const calculation = await getCalculationByPaymentRequestId(paymentRequestId)
+  const rawCalculation = await getCalculationByPaymentRequestId(paymentRequestId)
+  const calculation = rawCalculation.paymentRequestId ? rawCalculation : await updateCalculationPaymentRequestId(rawCalculation)
+
   const result = schema.validate(calculation, {
     abortEarly: false
   })
 
   if (result.error) {
+    console.log(calculation)
     throw new Error(`Payment request with paymentRequestId: ${paymentRequestId} does not have the required Calculation data: ${result.error.message}`)
   }
 
   return {

```

Comment:
`CalculationDate` isn't returned but is included in the select query and schema validation -- is is missing?